src/theory/bv/pb/pb_types.h

```diff
@@ -0,0 +1,160 @@
+/******************************************************************************
+ * Top contributors (to current version):
+ *   Alan Prado
+ *
+ * This file is part of the cvc5 project.
+ *
+ * Copyright (c) 2009-2024 by the authors listed in the file AUTHORS
+ * in the top-level source directory and their institutional affiliations.
+ * All rights reserved.  See the file COPYING in the top-level source
+ * directory for licensing information.
+ * ****************************************************************************
+ *
+ * Defines pseudo-Boolean types and utilities, including:
+ * - Enumerations for pseudo-Boolean values and solver states.
+ * - Classes for pseudo-Boolean variables and literals.
+ * - Classes for pseudo-Boolean constraints and constraint sets, with
+ *   constraints stored as Nodes to leverage hash consing.
+ */
+
+#include "cvc5_private.h"
+
+#ifndef CVC5__THEORY__BV__PB__PB_TYPES_H
+#define CVC5__THEORY__BV__PB__PB_TYPES_H
+
+#include <string>
+
+#include "expr/node.h"
+#include "util/integer.h"
+
+namespace cvc5::internal {
+namespace theory {
+namespace bv {
+namespace pb {
+
+class PbNodeManager;
+
+/**
+ * Represents the possible values of a Pseudo-Boolean variable.
+ * These values are used to describe the assignment or state of a variable.
+ */
+enum class PbValue
+{
+  UNKNOWN,
+  TRUE,
+  FALSE
+};
+
+/**
+ * Represents the possible states of a Pseudo-Boolean solver.
+ * These states indicate the outcome or current status of the solver.
+ */
+enum class PbSolveState
+{
+  UNKNOWN,
+  SAT,
+  UNSAT
+};
+
+/**
+ * A pseudo-Boolean variable identified by a unique ID.
+ */
+class PbVariable
+{
+ public:
+  explicit PbVariable(const Integer& id);
+  explicit PbVariable(const int& id);
+
+  friend std::ostream& operator<<(std::ostream& os, const PbVariable& var);
```

Comment:
Why are you declaring this method like this rather than globally in this namespace?